### ISSUE TITLE
 Error: error configuring x509 signer: decrypt: encrypted: unexpected kdf parameters

### ISSUE THREAD

**Issue by dlydiard** [tektoncd/chains#975]
Hello, I followed the directions for setting up Tekton Chains on https://docs.openshift.com/pipelines/1.12/secure/using-tekton-chains-for-openshift-pipelines-supply-chain-security.html#chains-signing-secrets-skopeo_using-tekton-chains-for-openshift-pipelines-supply-chain-security

i used 
```sh
export COSIGN_PASSWORD=$(openssl rand -base64 32 | tr -d "=+/")
cosign generate-key-pair k8s://openshift-pipelines/signing-secrets
```
using cosign v2.2.0. The `signing-secrets` `Secret` looks correct and contains the `cosign.key`, `cosign.password`, and `cosign.pub` keys.

to generate the secret and restarted the tekton-chains-controller.
I then ran my pipeline and the controller logs have the following errors:
```
{"level":"info","ts":"2023-10-28T00:13:29.376Z","logger":"watcher","caller":"x509/x509.go:195","msg":"Found cosign key...","commit":"6f5758f","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"0cf6cacd-1d34-4a4c-ad26-fc7040a63c7d","knative.dev/key":"bee-demo-pipeline/bee-demo-poc-3j04y9","pipelinerun":"bee-demo-pipeline/bee-demo-poc-3j04y9"}
{"level":"warn","ts":"2023-10-28T00:13:29.376Z","logger":"watcher","caller":"chains/signing.go:66","msg":"error configuring x509 signer: decrypt: encrypted: unexpected kdf parameters","commit":"6f5758f","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"0cf6cacd-1d34-4a4c-ad26-fc7040a63c7d","knative.dev/key":"bee-demo-pipeline/bee-demo-poc-3j04y9","pipelinerun":"bee-demo-pipeline/bee-demo-poc-3j04y9"}
{"level":"info","ts":"2023-10-28T00:13:29.376Z","logger":"watcher","caller":"chains/signing.go:147","msg":"Created payload of type in-toto for tekton.dev/v1beta1/PipelineRun bee-demo-pipeline/bee-demo-poc-3j04y9","commit":"6f5758f","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"0cf6cacd-1d34-4a4c-ad26-fc7040a63c7d","knative.dev/key":"bee-demo-pipeline/bee-demo-poc-3j04y9","pipelinerun":"bee-demo-pipeline/bee-demo-poc-3j04y9"}
{"level":"warn","ts":"2023-10-28T00:13:29.376Z","logger":"watcher","caller":"chains/signing.go:153","msg":"No signer x509 configured for tekton-pipeline-run","commit":"6f5758f","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"0cf6cacd-1d34-4a4c-ad26-fc7040a63c7d","knative.dev/key":"bee-demo-pipeline/bee-demo-poc-3j04y9","pipelinerun":"bee-demo-pipeline/bee-demo-poc-3j04y9"}
{"level":"info","ts":"2023-10-28T00:13:29.404Z","logger":"watcher","caller":"controller/controller.go:550","msg":"Reconcile succeeded","commit":"6f5758f","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"0cf6cacd-1d34-4a4c-ad26-fc7040a63c7d","knative.dev/key":"bee-demo-pipeline/bee-demo-poc-3j04y9","duration":0.034337237}
{"level":"info","ts":"2023-10-28T00:13:29.405Z","logger":"watcher","caller":"pipelinerun/pipelinerun.go:69","msg":"pipelinerun has been reconciled","commit":"6f5758f","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"cd7a89ce-5beb-4022-bd54-0814fbd3dc69","knative.dev/key":"bee-demo-pipeline/bee-demo-poc-3j04y9","pipelinerun":"bee-demo-pipeline/bee-demo-poc-3j04y9"}
{"level":"info","ts":"2023-10-28T00:13:29.405Z","logger":"watcher","caller":"controller/controller.go:550","msg":"Reconcile succeeded","commit":"6f5758f","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"cd7a89ce-5beb-4022-bd54-0814fbd3dc69","knative.dev/key":"bee-demo-pipeline/bee-demo-poc-3j04y9","duration":0.000087986}
```

chains is on v0.17.1

chains-config `ConfigMap`
```yaml
kind: ConfigMap
...
data:
  artifacts.taskrun.storage: tekton
  artifacts.pipelinerun.storage: tekton
  artifacts.pipelinerun.format: in-toto
  transparency.enabled: 'false'
  artifacts.taskrun.format: in-toto
  artifacts.oci.signer: x509
  artifacts.oci.storage: ''
  artifacts.pipelinerun.signer: x509
  artifacts.oci.format: simplesigning
  artifacts.taskrun.signer: x509
```

Any ideas on the log messages:
> Found cosign key...
> error configuring x509 signer: decrypt: encrypted: unexpected kdf 
> No signer x509 configured for tekton-pipeline-run

?

**Comment by lcarva** [comment]
@dlydiard , what version of Chains are you using? This looks a lot like https://github.com/tektoncd/chains/issues/918 which is fixed in Chains v0.18.0.

Older versions of Chains will only work with key-pairs generated from cosign v2.1.x and older.

**Comment by dlydiard** [comment]
using cosign v2.2.0 with chains v0.17.1. I downgraded cosign to 2.1.1 and it fixed the problem. ty!
